netsim/extra/repeater/plugin.py

```diff
@@ -0,0 +1,83 @@
+from box import Box
+from netsim import data
+from netsim.utils import log,strings
+from netsim.augment import links
+
+def clone_interfaces(link_data: Box, nodename: str, clones: list[str]) -> list[Box]:
+  """
+  update_ifindex - update any custom ifindex on the given interface
+  """
+  def update_ifindex(intf:Box, c:int) -> Box:
+    ifindex = intf.get('ifindex',None)
+    if ifindex is not None:
+      intf = data.get_box(intf)                                             # Need to make a fresh copy
+      intf.ifindex = ifindex + c + 1
+    return intf
+  
+  ifs = []
+  if nodename in [ i.node for i in link_data.interfaces ]:
+     link_data.pop('_linkname',None)
+     atts = [ i for i in link_data.interfaces if i.node==nodename ][0]      # Copy any interface atts
```

Comment:
This wouldn't work for nodes connected to themselves (someone loved to use them IIRC). It might be better to iterate over interfaces looking for the target nodename)

Reply:
We wouldn't want to disappoint our more sophisticated users - refactored